mikeio/dfsu.py

```diff
@@ -488,6 +499,119 @@ def _get_element_coords(self):
         self._ec = ec
         return ec
 
+    def contains(self, points):
+        import matplotlib.path as mp
+        try:
+            domain = self._shapely_domain2d
+        except:
+            warnings.warn('Could not determine if domain contains points. Failed to convert to_shapely()')
+            return None
+        
+        cnts = mp.Path(domain.exterior).contains_points(points)
+        for interj in domain.interiors:
+            in_hole = mp.Path(interj).contains_points(points) 
+            cnts = np.logical_and(cnts, ~in_hole)
+        return cnts
+
+
+    def _create_tree2d(self):
+        xy = self.geometry2d.element_coordinates[:,:2]
+        self._tree2d = cKDTree(xy)
+
+    def _find_n_nearest_2d_elements(self, x, y=None, n_points=1):        
+        if self._tree2d is None:
+            self._create_tree2d()
+
+        if y is None: 
+            p = x
+        else:
+            p = np.array((x, y))
+        d, elem_id = self._tree2d.query(p, k=n_points)
+        return elem_id, d
+
+    def get_overset_grid(self, dx=None, shape=None):
+        coords = self.node_coordinates   # node_ or element_
+        small = 1e-10        
+        x0 = coords[:,0].min() + small
+        y0 = coords[:,1].min() + small
+        x1 = coords[:,0].max() - small
+        y1 = coords[:,1].max() - small
+        xr = x1 - x0
+        yr = y1 - y0
+        
+        if (dx is None) and (shape is None):
+            if xr <= yr:
+                nx = 10
+                ny = int(np.ceil(yr/xr))*nx
+            else:
+                ny = 10
+                nx = int(np.ceil(xr/yr))*ny
+            dx = xr/(nx-1)
+            dy = yr/(ny-1)
+        else:
+            if dx is None:
+                nx, ny = shape
+                dx = xr/(nx-1)
+                dy = yr/(ny-1)
+            if shape is None:
+                dy = dx
```

Comment:
Allow dx to be either None, float or a two-item list or tuple to get different dx and dy

Reply:
Done